packages/themeables/core/src/index.web.ts

```diff
@@ -0,0 +1,29 @@
+import { createContext, useContext, createElement, FC } from 'react'
+import { component, startWithType } from 'refun'
+
+export type TThemeables<ThemeType, ComponentMappings> = { [key in keyof ComponentMappings]: (props: ComponentMappings[key]) => ThemeType }
+
+export const setupTheme = <ThemeType, ComponentMappings>(defaultTheme: TThemeables<ThemeType, ComponentMappings>) => {
+  const ThemePiece = createContext(defaultTheme)
+
+  type K = keyof ComponentMappings
+
+  const createThemeable = <P extends ThemeType>(name: K, Target: FC<any>) => {
+    const Themeable = component(
+      startWithType<Partial<P> & ComponentMappings[K]>(),
+      (props) => ({
+        ...useContext(ThemePiece)[name](props),
+        ...props,
+      })
+    )((props) => createElement(Target, props))
+
+    Themeable.displayName = `${name}`
```

Comment:
can name be something else than a `string` ? 🤔

Reply:
Yes, that's the symbol issue that I forgot to fix :D I'll add it thanks